components/next_steps_view/steps.ts

```diff
@@ -91,6 +93,16 @@ export const Steps: StepType[] = [
         component: InviteMembersStep,
         visible: true,
     },
+    {
+        id: RecommendedNextSteps.DOWNLOAD_APPS,
+        title: localizeMessage(
+            'next_steps_view.downloadDesktopAndMobile',
+            'Download the Desktop and Mobile apps',
+        ),
+        roles: [],
+        component: DownloadAppsStep,
+        visible: true,
```

Comment:
Would it make sense to add the condition of the treatment here instead of filtering in the selector below?

Reply:
I don't feel strongly either way. I lean 1/5 towards keeping the logic and setup as it is today.

That said, the current setup of defining the whole list of possibilities and filtering at the level of the "smart" redux-connected component is a little different than what I'd typically expect of redux passing down the info needed to filter, and the filtering happening within the component. Performance-wise, its memoized so I think its similar either way.

Reply:
0/5 just thinking it would be cleaner to keep it all within the step logic given that the filtering is step agnostic. Also, for future code cleanup if we decide to not go with this treatment.

Reply:
I might not be understanding what you mean by "keep it all within the step logic given that the filtering is step agnostic". What I _think_ you mean is do something like define `const Steps = [...]` and then on the next line do something like `if (hasTreatment) { Steps.push(theDownloadStep) }`.

Steps is defined outside a component so it doesn't have access to the data stored in redux or to any component props. So I think to change this we would need to either:
* place its definition within the component, and let the necessary data filter down from redux to build the list
* make a helper function that takes the args that are used in the selector function, which would be really similar to the selector function.

Reply:
Actually more something like: 
```suggestion
        visible: getDownloadAppsCTATreatment(state) === DownloadAppsCTATreatments.TIPS_AND_NEXT_STEPS,
```

Reply:
Ahh. That does seem like a good use for `visible` and it does keep it more step agnostic. But with Steps defined outside of a context that has access to redux `state`, I still think we'd need to place it in the component, in the selector function , or make a helper function.